jlm/rvsdg/region.hpp

```diff
@@ -87,24 +87,33 @@ class argument : public output
   structural_input * input_;
 };
 
+/**
+ * \brief Represents the result of a region.
+ *
+ * Region results represent the final values of the region's acyclic graph. The result values
+ * can be mapped back to the region arguments or the corresponding structural outputs
+ * throughout the execution, but the concrete semantics of this mapping
+ * depends on the structural node the region is part of. A region result is either linked
+ * with a \ref structural_output or is a standalone result.
+ */
 class result : public input
 {
-  jlm::util::intrusive_list_anchor<jlm::rvsdg::result> structural_output_anchor_;
+  util::intrusive_list_anchor<result> structural_output_anchor_;
 
 public:
-  typedef jlm::util::
-      intrusive_list_accessor<jlm::rvsdg::result, &jlm::rvsdg::result::structural_output_anchor_>
-          structural_output_accessor;
+  typedef util::intrusive_list_accessor<result, &result::structural_output_anchor_>
+      structural_output_accessor;
 
-  virtual ~result() noexcept;
+  ~result() noexcept override;
 
 protected:
   result(
-      jlm::rvsdg::region * region,
-      jlm::rvsdg::output * origin,
-      jlm::rvsdg::structural_output * output,
+      rvsdg::region * region,
+      rvsdg::output * origin,
```

Comment:
Why is 'rvsdg::' kept for 'region' and 'output', but not for 'structural_output'?
Are they used in other namespaces that cause conflicts?

Reply:
There is no conflict in terms of namespaces, but just potential confusion between type "output" and variable "output", if the namespace is removed. The problem is that the classes/methods have not been renamed to the CamelCase equivalent. That is why we still have this "problem". It will go away once we perform the renaming.